apps/backpack/views.py

```diff
@@ -45,24 +45,52 @@ def __init__(self, x, y, width, height, radius, text, strokecolor, fillcolor, st
         self.studyload = studyload
         self.esco = esco
 
+    def split_text(self, text, max_width):
+        words = text.split()
+        lines = []
+        current_line = ""
+
+        for word in words:
+            test_line = f"{current_line} {word}".strip()
+            if self.canv.stringWidth(test_line, 'Helvetica-Bold', 10) <= max_width:
+                current_line = test_line
+            else:
+                lines.append(current_line)
+                current_line = word
+
+        lines.append(current_line)
+        return lines    
+    
+
     def draw(self):
         self.canv.setFillColor(self.fillcolor)
         self.canv.setStrokeColor(self.strokecolor)
         self.canv.roundRect(self.x, self.y, self.width, self.height, self.radius,
                              stroke=1, fill=1)
         
         self.canv.setFillColor('#323232')
-        self.canv.setFont('Helvetica-Bold', 14)
-
         text_width = self.canv.stringWidth(self.text)
-        name = self.text
-        self.canv.drawString(self.x + 10, self.y + 15, name)
+        if text_width > self.width - 250:
+            self.canv.setFont('Helvetica-Bold', 10)
+            available_text_width = self.width - 150
+            y_text_position = self.y + 17.5
+        else:
+            self.canv.setFont('Helvetica-Bold', 14)
+            available_text_width = self.width - 250
+            y_text_position = self.y + 15
+
+        text_lines = self.split_text(self.text, available_text_width)
+
+        for line in text_lines:
+            self.canv.drawString(self.x + 10, y_text_position, line)
+            y_text_position -= 12 
         
         self.canv.setFillColor('blue')
         if self.esco:
-            self.canv.drawString(self.x + 10 + text_width, self.y + 15, " [E]")
-            # the whole rectangle links to esco, instead of just the "[E]"
-            self.canv.linkURL(f"http://data.europa.eu/{self.esco}", (self.x, self.y, self.width, self.height), relative=1, thickness=0)
+            last_line_width = self.canv.stringWidth(text_lines[-1])
+            self.canv.setFillColor('blue')
+            self.canv.drawString(self.x + 10 + last_line_width, y_text_position + 12, " [E]")
```

Comment:
Couldn't we just append the `" [E]"` to the string before already? Then we'd also make sure that the `[E]` doesn't overflow our boundaries...